test/websocket.js

```diff
@@ -0,0 +1,83 @@
+const assert = require("assert");
+const nodemailer = require("nodemailer");
+const request = require("supertest");
+const WebSocket = require("ws");
+
+const mailsacAPIKey = ""; // Generated by mailsac. See https://mailsac.com/api-keys
+const mailsacToAddress = "example@mailsac.com"; // Mailsac email address where the email will be sent
+const smtpUserName = ""; // Username for smtp server authentication
+const smtpPassword = ""; // Password for smtp server authentication
+const smtpHost = ""; // hostname of the smtp server
+const smtpPort = 587; // port the smtp is listening on
+
+const wait = (millis) => new Promise((resolve) => setTimeout(resolve, millis));
+
+describe("send email to mailsac", function () {
+  this.timeout(50000); // test can take a long time to run. This increases the default timeout for mocha
+
+  /* delete all messages in the inbox after the test runs to prevent leaky tests.
+       This requires the inbox to private, which is a paid feature of Mailsac.
+       The afterEach section could be omitted if using a public address
+    */
+  afterEach(() =>
+    request("https://mailsac.com")
+      .delete(`/api/addresses/${mailsacToAddress}/messages`)
+      .set("Mailsac-Key", mailsacAPIKey)
+      .expect(204)
+  );
+
+  it("sends email with link to example.com website", async () => {
+    // create a transporter object using the default SMTP transport
+    const transport = nodemailer.createTransport({
+      host: smtpHost,
+      port: smtpPort,
+      auth: {
+        user: smtpUserName,
+        pass: smtpPassword,
+      },
+    });
+    // send mail using the defined transport object
+    const result = await transport.sendMail({
+      from: smtpUserName, // sender address
+      to: mailsacToAddress, // recipient address
+      subject: "Hello!",
+      text: "Check out https://example.com",
+      html: "Check out <a href https://example.com>My website</a>",
+    });
+
+    // logs the messageId of the email, confirming the email was submitted to the smtp server
+    console.log("Sent email with messageId: ", result.messageId);
+
+    // Open websocket waiting for email. Use asynchronous wait to wait until email is received and processed by mailsac.
+    let messages = [];
+    const ws = new WebSocket(
+      `wss://sock.mailsac.com/incoming-messages?key=${mailsacAPIKey}&addresses=${mailsacToAddress}`
+    );
+    ws.on("open", () => {
+      console.log("web socket opened");
+    });
+    ws.on("error", (err) => {
+      console.log("connection error", err);
+    });
+    ws.on("message", (data) => {
```

Comment:
The benefit of using a websocket is to leverage event driven patterns. Not await every N seconds. May as well just hit the rest API. 

Try returning a new promise that resolves inside the 'message' event.

Reply:
Thanks for the feedback.  I adjusted the code to return a promise in the `message` event.